### PR TITLE
Binary support for ebd-all-minilm

### PR DESCRIPTION
**Pull request by nsosio**
Added shell script for building embedding binary

- replaces/closes #129
- part of https://github.com/premAI-io/prem-registry/issues/97

### CODE REVIEW

File: ebd-all-minilm/build-apple-darwin.sh
Comment (casperdcl) [inline ebd-all-minilm/build-apple-darwin.sh:1]: building is handled by:

https://github.com/premAI-io/prem-services/blob/8363a8ac97df313779c3a9748f21a0aa2bb7f473/.github/workflows/ci.yml#L41

so rename this file to `build-aarch64-apple-darwin.sh`

File: ebd-all-minilm/build-apple-darwin.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# Usage: ./build-apple-darwin.sh <arch> <arch_name>
+#
+# Parameters:
+#   arch: The architecture to use.
+#   arch_name: The architecture name to use.
+
+set -e
+
+ARCH=${1:-arm64}
+ARCH_NAME=${2:-aarch64}
```
Comment (casperdcl) [inline ebd-all-minilm/build-apple-darwin.sh:12]: would ignore intel for now :)

```suggestion
set -e
```

File: ebd-all-minilm/build-apple-darwin.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+export VERSION=1.0.5
+
+test -f venv/bin/activate || python -m venv venv
+source venv/bin/activate
+
+
+pip install -r requirements.txt pyinstaller
+NAME=ebd-all-minilm-${VERSION}-x86_64-apple-darwin
```
Comment (casperdcl) [inline ebd-all-minilm/build-apple-darwin.sh:21]:
```suggestion
NAME=ebd-all-minilm-${VERSION}-aarch64-apple-darwin
```

File: ebd-all-minilm/build-apple-darwin.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  --collect-data torch \
+  --copy-metadata torch \
+  --add-data 'main.py:.' \
+  --copy-metadata tokenizers \
+  --hidden-import=tiktoken_ext.openai_public \
+  --hidden-import=tiktoken_ext \
+  --target-arch $ARCH \
+  --name=$NAME \
+  main.py
+cp dist/$NAME dist/ebd-all-minilm-${VERSION%%.*}-$ARCH_NAME-apple-darwin
```
Comment (casperdcl) [inline ebd-all-minilm/build-apple-darwin.sh:42]:
```suggestion
cp dist/$NAME dist/ebd-all-minilm-${VERSION%%.*}-aarch64-apple-darwin
```

File: ebd-all-minilm/main.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import logging
 
+import tqdm  # noqa
```
Comment (biswaroop1547) [inline ebd-all-minilm/main.py:3]: we can remove this since we already doing `--copy-metadata` for this

File: ebd-all-minilm/build-apple-darwin.sh
Comment (nsosio) [reply ebd-all-minilm/build-apple-darwin.sh:1]: Done :)

File: ebd-all-minilm/build-apple-darwin.sh (hunk 1 above)
Comment (nsosio) [reply ebd-all-minilm/build-apple-darwin.sh]: Done :)

File: ebd-all-minilm/build-aarch64-apple-darwin.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ARCH_NAME=${2:-aarch64}
+
+export VERSION=1.0.5
+
+test -f venv/bin/activate || python -m venv venv
+source venv/bin/activate
+
+
+pip install -r requirements.txt pyinstaller
+NAME=ebd-all-minilm-${VERSION}-x86_64-apple-darwin
```
Comment (casperdcl) [inline ebd-all-minilm/build-aarch64-apple-darwin.sh:15]:
```suggestion
set -e

export VERSION=1.0.5

test -f venv/bin/activate || python -m venv venv
source venv/bin/activate

pip install -r requirements.txt pyinstaller
NAME=ebd-all-minilm-${VERSION}-aarch64-apple-darwin
```

File: ebd-all-minilm/main.py (hunk 4 above)
Comment (nsosio) [reply ebd-all-minilm/main.py]: Removed :)

File: ebd-all-minilm/build-aarch64-apple-darwin.sh (hunk 5 above)
Comment (nsosio) [reply ebd-all-minilm/build-aarch64-apple-darwin.sh]: Done :)

**Pull request merged**